crates/sparrow-interfaces/src/source.rs

```diff
@@ -21,12 +23,12 @@ pub trait Source: Send + Sync {
     fn read(
         &self,
         projected_datatype: &DataType,
-        read_config: ReadConfig,
-    ) -> BoxStream<'_, error_stack::Result<Batch, SourceError>>;
+        read_config: Arc<ReadConfig>,
+    ) -> BoxStream<'static, error_stack::Result<Batch, SourceError>>;
```

Comment:
Changed this to return `'static`. This allows the stream to be used without worrying about the lifetime of the source. This allows (eg.,) borrowing the source to start execution, and then getting a free-standing reader from it.

Reply:
@jordanrfrazier fyi